schemas/extend.js

```diff
@@ -0,0 +1,25 @@
+const extendRecordSchema = {
+  $id: '#/definitions/extendRecord',
+  type: 'object',
+  properties: {
+    function: {
+      type: 'string',
+      const: 'extendRecord',
+    },
+    name: {
+      type: 'string',
+      pattern: '^([a-zA-Z0-9][a-zA-Z0-9-]{0,49}[a-zA-Z0-9]|[a-zA-Z0-9]{1})$',
+    },
```

Comment:
we may want to consider creating generic individual schemas for repeated things like `name` and then just reference the schema instead of copy/pastaing everywhere.

Reply:
https://json-schema.org/understanding-json-schema/structuring.html#ref

start with just `name` and we can see how we like it.

Reply:
we can also do this in a separate PR